Cargo.toml

```diff
@@ -23,7 +23,7 @@ appveyor = { repository = "rust-random/rand" }
 [features]
 # Meta-features:
 default = ["std", "std_rng"]
-nightly = ["simd_support"] # enables all features requiring nightly rust
+nightly = ["simd_support", "partition_at_index"] # enables all features requiring nightly rust
```

Comment:
A simpler approach might be to just use the `nightly` feature instead of introducing a new one. This would have the advantage that we don't have to worry about dropping the feature in the future without breaking code.

Reply:
My reasoning here is that this allows leaving the feature flag in place even once "slice_partition_at_index" stabilizes, which would allow supporting older rustc versions. I also see your point about dropping the feature being a breaking change, though...

Reply:
Fair enough. I think we will likely just raise the minimum Rust version. @dhardy What do you think?

Reply:
In general, it is not possible to reliably support old nightlies, and not very useful either. So removing a feature flag only usable on nightly compilers is not an issue.